app/backend/core/authentication.py

```diff
@@ -216,3 +217,17 @@ async def get_auth_claims_if_enabled(self, headers: dict) -> dict[str, Any]:
             if self.require_access_control:
                 raise
             return {}
+
+    async def check_path_auth(self, path: str, auth_claims: dict[str, Any], search_client: SearchClient) -> bool:
+        # Start with the standard security filter for all queries
+        security_filter = self.build_security_filters(overrides={}, auth_claims=auth_claims)
+        # Filter down to only chunks that are from the specific source file
+        filter = f"{security_filter} and (sourcepage eq '{path}')"
+        print(f"path filter {filter}")
+
+        # If the filter returns any results, the user is allowed to access the document
+        # Otherwise, access is denied
+        results = await search_client.search(search_text="*", top=1, filter=filter, include_total_count=True)
```

Comment:
Doesn't include_total_count have a performance impact? (Docs say so). We only need to confirm that 1 matches, right?

Reply:
You are right - i can change this